app/controllers/application_controller.rb

```diff
@@ -55,7 +55,7 @@ def login_redirect?
   end
 
   def oauth_callback?
-    request.path == oauth_callback_path("google_oauth2")
+    request.path == oauth_callback_path(ENV['OAUTH_PROVIDER_NAME'])
```

Comment:
Same here, it would be good to call a config object rather than ENV vars deep inside controllers.